RuddockWebsite/modules/account/helpers.py

```diff
@@ -0,0 +1,79 @@
+from flask import g, flash
+from sqlalchemy import text
+from RuddockWebsite import auth_utils
+from RuddockWebsite import email_templates
+from RuddockWebsite import email_utils
+from RuddockWebsite import misc_utils
+from RuddockWebsite import validation_utils
+
+def get_user_data(user_id):
+  ''' Helper funciton to get user data. '''
+  query = text("""
+    SELECT fname, lname, uid, matriculate_year, grad_year, email
+    FROM members
+    WHERE user_id=:uid
+    """)
+  return g.db.execute(query, uid=user_id).first()
+
+def handle_create_account(user_id, username, password, password2, birthday):
+  '''
+  Creates a new account. Flashes a message and returns False if an error
+  occurs, otherwise returns True.
+  '''
+  # Validate username and password. The validate_* functions will flash errors.
+  # We want to check all fields and not just stop at the first error.
+  is_valid = True
+  if not validation_utils.validate_username(username):
+    is_valid = False
+  if not validation_utils.validate_password(password, password2):
+    is_valid = False
+  if not validation_utils.validate_birthday(birthday):
+    is_valid = False
+
+  if is_valid:
+    # Insert new values into the database. Because the password is updated in a
+    # separate step, we must use a transaction to execute this query.
+    trans = g.db.begin()
+    try:
+      # Insert the new row into users.
+      query = text("""
+        INSERT INTO users (user_id, username, password_hash)
+        VALUES (:uid, :u, :ph)
+        """)
+      g.db.execute(query, uid=user_id, u=username, ph='')
+      # Set the password.
+      auth_utils.set_password(username, password)
+      # Set the birthday.
+      query = text("""
+        UPDATE members
+        SET bday = :b
+        WHERE user_id = :u
+        """)
+      g.db.execute(query, b=birthday, u=user_id)
+      # Invalidate the account creation key.
+      query = text("""
+        UPDATE members
+        SET create_account_key = NULL
+        WHERE user_id = :u
+        """)
+      g.db.execute(query, u=user_id)
+      trans.commit()
+    except Exception:
+      trans.rollback()
```

Comment:
It works like normal database transactions - nothing is changed in the database until `trans.commit()`, at which point all the changes happen at once. I basically just followed the documentation on how to use it here:

http://docs.sqlalchemy.org/en/rel_0_8/core/connections.html#using-transactions

I do not know if `trans.commit()` can raise exceptions (it shouldn't unless something very strange happens, like the db connection disappears), but using transactions like this at least guarantees that we won't run into issues where a user's account is partially created and then interrupted.